gui/syncthing/core/eventService.js

```diff
@@ -18,14 +28,42 @@ angular.module('syncthing.core')
                 return;
             }
             $rootScope.$broadcast(self.ONLINE);
-
+            
+            
             if (lastID > 0) {   // not emit events from first response
+                var greenCounter = 0;
+                var redCounter = 0;
+                
                 data.forEach(function (event) {
                     if (debugEvents) {
                         console.log("event", event.id, event.type, event.data);
                     }
                     $rootScope.$broadcast(event.type, event);
+                    
+                    // Counters for favicon change decision
+                    if (event.type == self.FOLDER_ERRORS) {
+                        redCounter++;
+                    }
+                    if (event.type == self.ITEM_STARTED || event.type == self.DOWNLOAD_PROGRESS) {
+                        greenCounter++;
+                    }
+                    if (event.type == self.STATE_CHANGED && event.data.to == 'syncing') {
+                        greenCounter++;
+                    }
+                    
```

Comment:
maybe this could be simplified with a default state (see above) and then change the state accordingly, e.g.

    if (event.type == self.FOLDER_ERRORS) {
      iconstate = 'error';
    }
    elseif (other conditions) {
      ...
    }
    changeFavIcon(iconstate);

this would make the error state superseed the rest, then the next, etc. etc. - no further counting needed